File: war-away-fe/src/app/model/offer.ts
import {User} from "./user";

export interface IOffer {
  id: number;
  user: User;
  phoneNumber: string;
  departurePoint: string;
  arrivalPoint: string;
  departureDate: Date;
  originLat: string;
  originLong: string;
  destLat: string;
  destLong: string;
  distance: string;
  personNumber: number;
  pets: string;
  description: string;
}

export class Offer implements IOffer {
  constructor(
    public id: number,
    public user: User,
    public phoneNumber: string,
    public departurePoint: string,
    public arrivalPoint: string,
    public departureDate: Date,
    public originLat: string,
    public originLong: string,
    public destLat: string,
    public destLong: string,
    public distance: string,
    public personNumber: number,
    public pets: string,
    public description: string
  ){
    this.id = id;
    this.user = user;
    this.phoneNumber = phoneNumber;
    this.departurePoint = departurePoint;
    this.arrivalPoint = arrivalPoint;
    this.departureDate = departureDate;
    this.originLat = originLat;
    this.originLong = originLong;
    this.destLat = destLat;
    this.destLong = destLong;
    this.distance = distance;
    this.personNumber = personNumber;
    this.pets = pets;
    this.description = description
  }
}

File: waraway/src/main/java/war/away/waraway/infrastructure/entities/Offer.java
package war.away.waraway.infrastructure.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.*;
import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import java.util.Date;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity(name = "offer")
@Table(name = "offer")
public class Offer {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    @JsonBackReference
    private User user;

    @Column(length = 150)
    private String phoneNumber;

    @Column(nullable = false, length = 150)
    private String departurePoint;

    @Column(length = 150)
    private String arrivalPoint;

    @Column(length = 150)
    @DateTimeFormat(pattern = "dd-MM-yyyy")
    @Temporal(TemporalType.DATE)
    private Date departureDate;

    @Column()
    private String originLat;

    @Column()
    private String originLong;

    @Column()
    private String destLat;

    @Column()
    private String destLong;

    @Column()
    private String distance;

    @Column()
    private Integer personNumber;

    @Column()
    private Boolean pets;

    @Column()
    private String description;
}
